optuna/samplers/_tpe/sampler.py

```diff
@@ -814,8 +816,8 @@ def _calculate_weights_below_for_multi_objective(
         contributions = np.asarray(
             [hv - WFG().compute(lvals[indices_mat[i]], reference_point) for i in range(n_below)]
         )
-        contributions += EPS
-        weights_below = np.clip(contributions / np.max(contributions), 0, 1)
+        contributions = np.clip(contributions, EPS, None)
+        weights_below = np.clip(contributions / np.max(contributions), EPS, 1.0)
```

Comment:
If the debugging is the purpose, what about the following and removing the line in `_build_parzen_estimator`?
```suggestion
        contributions[np.isnan(contributions)] = np.inf
        max_contribution = np.max(contributions)
        if not np.isfinite(max_contribution):
            weights_below = np.ones_like(contributions, dtype=float)
        else:
            weights_below = np.clip(contributions / max_contribution, EPS, 1)
```

Reply:
I thought `contributions = np.clip(contributions, EPS, None)` is slightly better than `contributions += EPS` just because `contributions` can be close to `sys.float_info.max`; in that case, adding eps results inf.
However, since that rarely happens, I reconsidered and agree that your suggestion is better.